### PR TITLE
Character's skills

### PR DESCRIPTION
**Pull request by Anapaotrev**
### What does this PR do ?
* Creates the skills model
* Adds the skills to the character
bitmoji

### CODE REVIEW

File: app/models/skill.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Skill < ApplicationRecord
+  belongs_to :character
+
+  validates :name, :description, :difficulty, :characteristic, :type_of_skill
+
+  validates :rank,
+  			numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 5 }
+
+  def parse_skills
```
Comment (kurenn) [inline app/models/skill.rb]: This should be a class method...

File: app/models/character.rb [hunk 2]
```diff
@@ -1,3 +1,17 @@
 class Character < ApplicationRecord
   belongs_to :user
-end
+
+  has_many :skills, dependent: :destroy
+
+  after_create :assign_skills
+
+  def assign_skills
+  	Skill.new.parse_skills.each do |skill|
```
Comment (kurenn) [inline app/models/character.rb]: So you need to instantiate a skill to create other skills?

File: spec/models/character_spec.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe Character, type: :model do
```
Comment (kurenn) [inline spec/models/character_spec.rb]: There are tests missing for validations

File: spec/models/skill_spec.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Skill, type: :model do
```
Comment (kurenn) [inline spec/models/skill_spec.rb:3]: A good test to add would be the `parse_skills` method

**Review by kurenn — CHANGES_REQUESTED** [review]

File: spec/models/skill_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  it { should validate_presence_of(:characteristic) }
+  it { should validate_presence_of(:type_of_skill) }
+  it { should validate_presence_of(:difficulty) }
+  it { should validate_presence_of(:description) }
+  it { should validate_numericality_of(:rank).is_greater_than_or_equal_to(0).is_less_than_or_equal_to(5) }
+
+
+end
+
+describe "Skill" do
```
Comment (kurenn) [inline spec/models/skill_spec.rb]: Why is this outside the Skill block?

File: spec/models/skill_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  it { should validate_presence_of(:description) }
+  it { should validate_numericality_of(:rank).is_greater_than_or_equal_to(0).is_less_than_or_equal_to(5) }
+
+
+end
+
+describe "Skill" do 
+	describe "parse_skills" do
+		it "should return all the skills of the jason" do 
+			file = File.read("#{Rails.root}/lib/data/Skills.json")
```
Comment (kurenn) [inline spec/models/skill_spec.rb]: shits-on-fire-yo_o_153652

File: spec/models/skill_spec.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+		  	data_hash = JSON.parse(file)
+		  	hash_of_skills = data_hash["Skills"]["Skill"]
```
Comment (kurenn) [inline spec/models/skill_spec.rb]: This test is too valid as you are using the actual implementation to test the output of the test...

File: spec/models/character_spec.rb [hunk 8]
```diff
@@ -36,4 +36,4 @@
   it { should validate_presence_of(:motivations) }
   it { should validate_presence_of(:species) }
   it { should validate_presence_of(:campaign) }
```
Comment (kurenn) [inline spec/models/character_spec.rb:38]: Are you missing a skills has  many test?

**Review by kurenn — CHANGES_REQUESTED** [review]

File: spec/models/skill_spec.rb (hunk 6 above)
Comment (Anapaotrev) [reply spec/models/skill_spec.rb]: bitmoji

File: spec/models/skill_spec.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  let(:skill) { FactoryGirl.create :skill }
+
+  	it { should validate_presence_of(:name) }
+  	it { should validate_presence_of(:characteristic) }
+ 	it { should validate_presence_of(:type_of_skill) }
+  	it { should validate_presence_of(:difficulty) }
+  	it { should validate_presence_of(:description) }
+  	it { should validate_numericality_of(:rank).is_greater_than_or_equal_to(0).is_less_than_or_equal_to(5) }
+  	
+	describe "Skill" do
```
Comment (kurenn) [inline spec/models/skill_spec.rb]: Another `describe "Skill"` isn't up there?

**Review by kurenn — CHANGES_REQUESTED** [review]

File: app/services/json_skill.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
+class JsonSkill
```
Comment (kurenn) [inline app/services/json_skill.rb:1]: 9b62f01e-3953-11e7-80f8-797f7f08a43e

**Review by kurenn — APPROVED** [review]

File: app/services/json_skill.rb (hunk 10 above)
Comment (Anapaotrev) [reply app/services/json_skill.rb:1]: bitmoji

**Pull request merged**